kornia/geometry/line.py

```diff
@@ -0,0 +1,173 @@
+# kornia.geometry.line module inspired by Eigen::geometry::ParametrizedLine
+# https://gitlab.com/libeigen/eigen/-/blob/master/Eigen/src/Geometry/ParametrizedLine.h
+from typing import Optional, Union
+
+import torch
+
+from kornia.core import Module, Parameter, Tensor, normalize
+from kornia.geometry.linalg import squared_norm
+from kornia.testing import KORNIA_CHECK, KORNIA_CHECK_IS_TENSOR, KORNIA_CHECK_SHAPE
+
+__all__ = ["ParametrizedLine", "fit_line"]
+
+
+# TODO: implement me: https://gitlab.com/libeigen/eigen/-/blob/master/Eigen/src/Geometry/Hyperplane.h
+class Hyperplane:
+    """Not implemented yet: https://gitlab.com/libeigen/eigen/-/blob/master/Eigen/src/Geometry/Hyperplane.h."""
+
+    pass
+
+
+class ParametrizedLine(Module):
+    """Class that describes a parametrize line.
+
+    A parametrized line is defined by an origin point :math:`o` and a unit
+    direction vector :math:`d` such that the line corresponds to the set
+
+    .. math::
+
+        l(t) = o + t * d
+    """
+
+    def __init__(self, origin: Tensor, direction: Tensor) -> None:
+        """Initializes a parametrized line of direction and origin.
+
+        Args:
+            origin: the origin point of the line of any dimension.
+            direction: the normalized vector direction of any dimension.
```

Comment:
why must it be normalized?

Reply:
i guess a design choice from Eigen

Reply:
the only reason I see is to compute distances. Do we want to force it?

Reply:
i would sitck to Eigen o have compatibility. LAter we can make improvements based on user feedback